### DISCUSSION TITLE
Upgrading from 4.2.100 to 5.2.100 fails on windows server 2016 64bit

### DISCUSSION THREAD

**Discussion by garrmark** [ravendb/ravendb#13720] [Q&A] [answered]
I have a 3 node cluster running in windows, I take down one node and replace the binaries with the 5.2.100 binaries, but the code crashes on startup with 

```
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
  <System>
    <Provider Name="Application Error"/>
    <EventID Qualifiers="0">1000</EventID>
    <Level>2</Level>
    <Task>100</Task>
    <Keywords>0x80000000000000</Keywords>
    <TimeCreated SystemTime="2022-03-01T17:54:45.648405900Z"/>
    <EventRecordID>22930019</EventRecordID>
    <Channel>Application</Channel>
    <Computer>EXGVSRV019.mypartssource.com</Computer>
    <Security/>
  </System>
  <EventData>
    <Data>Raven.Server.exe</Data>
    <Data>5.2.100.52022</Data>
    <Data>61954b73</Data>
    <Data>ntdll.dll</Data>
    <Data>10.0.14393.4886</Data>
    <Data>61d52226</Data>
    <Data>c0000409</Data>
    <Data>000000000009603f</Data>
    <Data>384c</Data>
    <Data>01d82d956e651ca8</Data>
    <Data>E:\RavenDB\Server\Raven.Server.exe</Data>
    <Data>C:\Windows\SYSTEM32\ntdll.dll</Data>
    <Data>7df0b94b-c58f-40c0-baf3-fb3593933ecf</Data>
    <Data/>
    <Data/>
  </EventData>
</Event>
```

Do I need to do an intermediate upgrade?  How can I get more information on the issue?

**Comment by ppekrol** [comment]
Hi @garrmark 

I'm assuming that you are running RavenDB as a Service? This event does not tell a lot. Can you try running the Raven.Server.exe from cmd and check the error message then?

**Reply by garrmark** [reply]
Yea, I get the exact same result, just pops a dialog saying Raven.Server has stopped working and gives that event log. There is a windows error reporting folder created, let me see what's in there.

**Reply by ppekrol** [reply]
Can you unpack the 5.2.100 DLLs to some temp folder, execute 'run.ps1', go through the setup (unsecured) to check if empty server starts at all?

**Reply by garrmark** [reply]
I get the same crash with an empty server.

**Reply by garrmark** [reply]
Here is the list of my installed software:
```
PS E:\RavenDB-5.2.100-windows-x64> wmic
wmic:root\cli>product get name
Name
New Relic CLI
Puppet (64-bit)
Microsoft Visual C++ 2010  x64 Redistributable - 10.0.40219
Microsoft .NET Core Host FX Resolver - 3.1.4 (x64)
Microsoft Visual C++ 2010  x86 Redistributable - 10.0.40219
Microsoft Visual C++ 2013 x86 Minimum Runtime - 12.0.21005
Microsoft Visual C++ 2013 x86 Additional Runtime - 12.0.21005
Trend Micro Security Agent
Dexterity Shared Components 11.0 (64-bit)
Node.js
Microsoft Visual C++ 2019 X64 Additional Runtime - 14.28.29913
Debug Diagnostics 2 Update 2
Microsoft ASP.NET Core 3.1.4 Shared Framework (x86)
Microsoft .NET Core Runtime - 3.1.4 (x64)
Microsoft .NET Core Runtime - 3.1.4 (x86)
SAP Crystal Reports runtime engine for .NET Framework (64-bit)
CrowdStrike Firmware Analysis
Microsoft Visual C++ 2008 Redistributable - x64 9.0.30729.6161
Adobe Reader 9.5.0
Microsoft Visual C++ 2012 x86 Additional Runtime - 11.0.51106
Microsoft Visual C++ 2008 Redistributable - x86 9.0.30729.6161
Microsoft .NET Core Host FX Resolver - 3.1.4 (x86)
VMware Tools
Microsoft Visual C++ 2019 X64 Minimum Runtime - 14.28.29913
Microsoft Visual C++ 2012 x64 Additional Runtime - 11.0.61030
Microsoft .NET Core Host - 3.1.4 (x86)
Microsoft SQL Server 2012 Native Client
Google Update Helper
Microsoft Visual C++ 2019 X86 Minimum Runtime - 14.26.28720
NSClient++ (x64)
Dexterity Shared Components 16.0 (64-bit)
Microsoft Visual C++ 2019 X86 Additional Runtime - 14.26.28720
CrowdStrike Sensor Platform
Microsoft SQL Server 2008 Native Client
Azure Connected Machine Agent
Microsoft Visual C++ 2012 x86 Minimum Runtime - 11.0.51106
Microsoft Visual C++ 2005 Redistributable
Microsoft Visual C++ 2012 x64 Minimum Runtime - 11.0.61030
Microsoft ASP.NET Core 3.1.4 Shared Framework (x64)
Automox Agent
CrowdStrike Device Control
Microsoft Visual C++ 2008 Redistributable - x86 9.0.30729.4148
eConnect for Microsoft Dynamics GP 2016
New Relic Infrastructure Agent
Dynatrace Agent 7.0
Google Chrome
IBM Data Server Client - DB2COPY1
Microsoft .NET Core Host - 3.1.4 (x64)
```

**Reply by ppekrol** [reply]
We have updated to .NET 6, so I'm suspecting this, but we ourselves do have WS2016 x64 and there are no issues there. Also checked the .NET 6 requirements and I do not see anything about WS2016 not being supported: https://github.com/dotnet/core/blob/main/release-notes/6.0/supported-os.md

Is your OS up to date? All windows updates are installed? Can you install latest vc_redist x64 https://aka.ms/vs/17/release/vc_redist.x64.exe and x86 https://aka.ms/vs/17/release/vc_redist.x86.exe ? Official page: https://docs.microsoft.com/en-us/cpp/windows/latest-supported-vc-redist?view=msvc-170

**Reply by garrmark** [reply]
we are up to date on windows updates, I will try the VC redist.

**Reply by garrmark** [reply]
Installing the latest vc_redist didn't fix the issue, still crashing

**Reply by ppekrol** [reply]
We will try to reproduce that locally. Which WS 2016 you have exactly? x64 Datacenter/Standard/Essentials?

**Reply by garrmark** [reply]
Datacenter

**Reply by garrmark** [reply]
Thanks you for your time on this, I have found the issue, we had a dynatrace agent that was injecting code into the executable, when I removed the dynatrace agent, it resolved the crash.

**Comment by garrmark** [comment, marked as answer]
I found the issue, the Dynatrace Agent 7.0 that we had installed on the server was causing the crash, we no longer use this agent, so I uninstalled it and am working now, thanks @ppekrol for your time and effort.

**Reply by ppekrol** [reply]
Great news, glad that this is sorted. I'm not familiar with Dynatrace, but it looks like it tried to attach to the .NET process for monitoring purposes and it lead to crash. I see in latest changelogs of sth that they are calling OneAgent few fixes around crashes with .NET processes.

**Reply by simendsjo** [reply]
We started getting crashes beginning of April 2023. Disabled deep monitoring of the application, and the crashes stopped. Not the first time we've experienced bugs in the code injected by Dynatrace causing crashes.
